### DISCUSSION TITLE
Help needed with CMake and modules

### DISCUSSION THREAD

**Discussion by alfredh** [baresip/baresip#1971] [Announcements]
Hi,

We are in the process of adding support for CMake, and part of this process is to upgrade
around 90 modules to add support for CMake.

If you are module maintainer or have some extra time, please consider helping out with
adding a CMakeLists.txt file to a random module.

Alfred

**Comment by alfredh** [comment]
these modules are next on my list:

ctrl_dbus @cspiel1 
ctrl_tcp @jmillan 
directfb Gary Metalle

if you have some extra time, please help to add support for CMake in your module

**Comment by alfredh** [comment]
We are having good progress on adding CMake to modules.
Many thanks to people who contribute.


Here are some modules next on my list:

- directfb
- g7221
- gst
- httpreq
- jack
- l16


any help is appreciated!

**Reply by cspiel1** [reply]
We take `gst` and `httpreq`.

**Comment by alfredh** [comment]
we have two modules webrtc_aec and webrtc_aecm that are doing almost the same thing.

Should we try to merge them into one module ?


The same goes for "pulse" and "pulse_async" modules.

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> we have two modules webrtc_aec and webrtc_aecm that are doing almost
> the same thing.

I'm using webrtc_aecm in my Android app using webrtc source from
pulseaudio project.  I have done quite a bit of work to get Android
build of the module working.  In order to avoid new hassle, I would like
to keep them separate.

**Reply by alfredh** [reply]
okay, thanks for the info. I did a diff of the two modules, they are using different API
and also different versions of libwebrtc.

Are you also using `webrtc_aecm` on Desktop ?

I think for longterm we should try to have only module that can use both regular mode and Mobile mode ...

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> Are you also using `webrtc_aecm` on Desktop ?

No, on Debian I use webrtc_aec.

> I think for longterm we should try to have only module that can use
> both regular mode and Mobile mode ...

Yes.

**Comment by cspiel1** [comment]
`pulse_async` should bring benefits for low latency.
`pulse` will become obsolete and replaced by `pulse_async`. But currently we keep it for more stability. We can drop it in a few month. After more tests with `pulse_async`.

Shortly we improved the pulseaudio reconnect procedure. E.g. if pulseaudio is restarted.

**Comment by sreimers** [comment]
Good progress, only a few left, I think we can try to replace `Makefile` with a simple CMake wrapper (like retest) for `v2.8.0`, so `v2.7.x` (released tomorrow) could be the last native Makefile release:

```
./i2s
./gtk
./webrtc_aec
./webrtc_aecm
./portaudio
./netroam
./jack
./mpa
./opus_multistream
./sdl
./opensles
./sndio
./gst
```

**Reply by alfredh** [reply]
I think these modules could also be removed:

- i2s
- mpa
- opus_multistream

**Reply by cspiel1** [reply]
We used i2s audio backend once for an Espressif esp32 phone. I have it in a branch on my github clone of baresip. It might not be up to date with newest  ausrc/auplay API. We can remove it for now.

**Comment by alfredh** [comment]
Many thanks to everyone who helped out with adding CMake files!

I would like to suggest now that we switch the Github CI (Workflows)
from GNU Make to CMake only.

The following projects have CI that needs updating to CMake:

- re
- rem
- retest
- baresip

After this step we can consider CMake as the default build system.
